### PR TITLE
Updates NAMESERVER15

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

The objective of the test case states about name servers that "it may sometimes be desirable not to reveal that information" but then if it does, this test case only sends an INFO message. An INFO message does not signal that there is any reason to look at it. This PR raises the level to NOTICE.

```
$ zonemaster-cli --test nameserver/nameserver15 telia.com --raw
   0.00 INFO     UNSPECIFIED    GLOBAL_VERSION   version=v4.7.2
   1.57 INFO     NAMESERVER15   N15_SOFTWARE_VERSION   ns_ip_list=dns358.fi.telia.net/88.194.40.67;dns49.de.telia.net/2001:2030:c000:5::4;dns49.de.telia.net/213.248.77.82; query_name=version.bind; string=9.18.9
   1.58 INFO     NAMESERVER15   N15_SOFTWARE_VERSION   ns_ip_list=dns1.telia.com/2001:2040:c001:101::5;dns1.telia.com/81.228.11.67;dns2.telia.com/2001:2040:c001:401::6;dns2.telia.com/81.228.10.67; query_name=version.bind; string=9.18.11
```

The specification does not distinguish between a server that does not respond at all on a "version query" and a server that responds, but with no version information. In the following example some of the servers do not respond at all and some of them responds with no information. This PR splits the message and make the test case make a distinction.
```
$ zonemaster-cli --test nameserver/nameserver15 festo.press --raw
   0.00 INFO     UNSPECIFIED    GLOBAL_VERSION   version=v4.7.2
  41.67 INFO     NAMESERVER15   N15_NO_VERSION   ns_ip_list=ns2.atvirtual.com/185.136.96.210;ns2.atvirtual.com/2a06:fb00:1::1:210;ns6.atvirtual.com/185.136.97.210;ns6.atvirtual.com/2a06:fb00:1::2:210;ns7.atvirtual.com/185.136.98.210;ns7.atvirtual.com/2a06:fb00:1::3:210;ns8.atvirtual.com/185.136.99.210;ns8.atvirtual.com/2a06:fb00:1::4:210
```

This PR also make the test case skip servers that do not even answer SOA queries to make it lift up any servers that do not answer because it is a "version query".

This PR also changes the argument from `ns_ip_list` to `ns_list` to give more information. It uses the same technique introduced by #1179.

This PR also updates the test case to check the class in the response.

## Context

Also see https://github.com/zonemaster/zonemaster-engine/issues/1281

Also see #1208

Also see test zones that match this update in #1217.

## Changes

The specification of NAMESERVER15.

## How to test this PR

Review the proposed changes.

### CODE REVIEW

**Comment by matsduf** [comment]
@tgreenx, you wrote the specification. Please review this proposal of updates.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md:40]: The suggested change requires more changes. I do not think it is worth it.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md]: I have moved it.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md]: I have completely changed the logic.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md]: Updated differently.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md]: I did the change anyway.

**Review by marc-vanderwal — APPROVED** [review]

**Comment by matsduf** [comment]
@tgreenx and @marc-vanderwal, please re-review.

File: docs/public/specifications/tests/Nameserver-TP/nameserver15.md
Comment (marc-vanderwal) [inline docs/public/specifications/tests/Nameserver-TP/nameserver15.md:95]: Just one suggestion, in order to address #1208 properly:
```suggestion
          1. If [DNS Class] of the TXT record is not CH, then add the name server
             to the *Wrong Record Class* set and go to next TXT record.
```

Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md:95]: Since that is in a loop "for each TXT record" what would "go to next TXT record" add?

Comment (matsduf) [reply docs/public/specifications/tests/Nameserver-TP/nameserver15.md:95]: Sometimes there is too much magic behind the scene. The idea is to just continue if the TXT record is in, say, IN class. That should work, should it? Unless some code behind filters non-CH responses.

**Comment by matsduf** [comment]
@marc-vanderwal, are you also fine with this?

**Review by marc-vanderwal — DISMISSED** [review]
Yes, that looks good to me!

**Comment by matsduf** [comment]
Test zone specification for this test case in #1217

**Comment by matsduf** [comment]
@tgreenx and @marc-vanderwal, I a version string just consisting of space to be equal to empty version string.

**Comment by matsduf** [comment]
> Test zone specification for this test case in #1217

Now the test zones have been implemented.

**Review by marc-vanderwal — DISMISSED** [review]

**Comment by matsduf** [comment]
@tgreenx, you did the current implementation of Nameserver15. Could you do the update?

**Comment by matsduf** [comment]
A trivial merge conflict was resolved after two approvals. Merge without waiting for renewed approval.

**Pull request merged**
